app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -58,22 +58,17 @@
 
     private final static String TAG = "SmsDetector";
 
-    private DataInputStream dis;
     private DataOutputStream dos;
+    BufferedReader in;
     private AimsicdService mAimsicdService;
-    private SharedPreferences prefs;
     private boolean mBound;
     private AIMSICDDbAdapter dbacess;
     private Context tContext;
-    private String[] SILENT_ONLY_TAGS;
+    private String[] LOADED_DETECTION_STRINGS;
+    private int B_LINED_SIZE = 30;// how many previous lines to hold
+    private String[] BUFFEREDLINES = new String[B_LINED_SIZE];//for holding previous lines from logcat that hold data we need
```

Comment:
if value/content of those fields won't change, make them private static final to save memory and prevent changing them by mistake.